php/class-cache.php

```diff
@@ -0,0 +1,1019 @@
+			$theme_location = $theme->get_stylesheet_directory();
+			$theme_slug     = basename( dirname( $theme_location ) ) . '/' . basename( $theme_location );
+			$slug           = sanitize_file_name( $theme_slug );
+			$rows[ $slug ]  = array(
+				'title'    => $theme->get( 'Name' ),
+				'url'      => $theme->get_stylesheet_directory_uri(),
+				'src_path' => $theme->get_stylesheet_directory(),
+			);
+		}
+
+		return array(
+			'slug'       => 'theme_files',
+			'type'       => 'folder_table',
+			'title'      => __( 'Theme', 'cloudinary' ),
+			'root_paths' => $rows,
+		);
+	}
+
+	/**
+	 * Get the settings structure for the WordPress table.
+	 *
+	 * @return array
+	 */
+	protected function get_wp_table() {
+
+		$rows = array();
+		// Admin folder.
+		$rows['wp_admin'] = array(
+			'title'    => __( 'WordPress Admin', 'cloudinary' ),
+			'url'      => admin_url(),
+			'src_path' => $this->file_system->wp_admin_dir(),
+		);
+		// Includes folder.
+		$rows['wp_includes'] = array(
+			'title'    => __( 'WordPress Includes', 'cloudinary' ),
+			'url'      => includes_url(),
+			'src_path' => $this->file_system->wp_includes_dir(),
+		);
+
+		return array(
+			'slug'       => 'wordpress_files',
+			'type'       => 'folder_table',
+			'title'      => __( 'WordPress', 'cloudinary' ),
+			'root_paths' => $rows,
+		);
+	}
+
+	/**
+	 * Get the settings structure for the WordPress table.
+	 *
+	 * @return array
+	 */
+	protected function get_content_table() {
+
+		$rows               = array();
+		$uploads            = wp_get_upload_dir();
+		$rows['wp_content'] = array(
+			'title'    => __( 'Uploads', 'cloudinary' ),
+			'url'      => $uploads['baseurl'],
+			'src_path' => $uploads['basedir'],
+		);
+
+		return array(
+			'slug'       => 'content_files',
+			'type'       => 'folder_table',
+			'title'      => __( 'Content', 'cloudinary' ),
+			'root_paths' => $rows,
+		);
+	}
+
+	/**
+	 * Setup the cache object.
+	 */
+	public function setup() {
+		$this->setup_setting_tabs();
+		$this->cache_point->init();
+	}
+
+	/**
+	 * Adds the individual setting tabs.
+	 */
+	protected function setup_setting_tabs() {
+		$cache_settings = $this->get_cache_settings();
+		foreach ( $cache_settings as $setting ) {
+			$callback = $setting->get_param( 'callback' );
+			if ( is_callable( $callback ) ) {
+				call_user_func( $callback ); // Init the settings.
+			}
+		}
+	}
+
+	/**
+	 * Get all the Cache settings.
+	 *
+	 * @return Setting[]
+	 */
+	public function get_cache_settings() {
+		static $settings = array();
+		if ( empty( $settings ) ) {
+			$main_setting = $this->settings->get_setting( 'cache_paths' );
+			foreach ( $main_setting->get_settings() as $slug => $setting ) {
+				$settings[ $slug ] = $setting;
+			}
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * Check to see if cache setting is enabled.
+	 *
+	 * @param string $cache_setting The setting slug to check.
+	 *
+	 * @return bool
+	 */
+	protected function is_cache_setting_enabled( $cache_setting ) {
```

Comment:
@DavidCramer 

I don't think this logic is wired up. I can't disable the cache :)